include/bluetooth/mesh/access.h

```diff
@@ -166,8 +166,10 @@ struct bt_mesh_model_op {
 	/** OpCode encoded using the BT_MESH_MODEL_OP_* macros */
 	const uint32_t  opcode;
 
-	/** Minimum required message length */
-	const size_t min_len;
+	/** Required message length. If the message has variable length then set
+	 * to -1 and perform length check in handler function.
+	 */
+	const size_t len;
```

Comment:
`size_t` is an unsigned type (`ssize_t` would be the signed variant), so assigning `-1` values to it doesn't seem right.

Additionally, this will introduce silent breakage (not caught at build time) to any out-of-tree Mesh models that have message handlers expecting variable-length payloads. At the bare minimum such a breakage needs to be documented in the release notes, however I would instead consider extending the `bt_mesh_model_op` struct with a new boolean member to indicate whether the given length is a precise value or a minimum value ('false' would have to mean "minimum" in order to maintain backwards compatibility with out-of-tree models)

Reply:
Just brainstorming here (might be a crazy idea), but if we don't want to increase the memory footprint by adding another member, we could consider some macros like this as well:
```
#define BT_MESH_LEN_VAR(len) (len)
#define BT_MESH_LEN_FIX(len) (SIZE_T_MAX - (len))
```

And then in `access.c` check if the length is above some `SIZE_T_MAX - BT_MESH_MAX_LEN` and perform the conversion if necessary. This would also preserve backwards compatibility. Does that seem too hackish? :)

Reply:
...or change the member type to `ssize_t` and use negative values to indicate precise lengths. (so same idea but just different approach)

Reply:
I like the one with `ssize_t`.